src/Cassandra/Requests/RequestExecution.cs

```diff
@@ -501,47 +501,61 @@ private void PrepareAndRetry(byte[] id, Host host)
                     .SetKeyspace(preparedKeyspace)
                     .ContinueSync(_ =>
                     {
-                        Send(request, host, ReprepareResponseHandler);
+                        Send(request, host, NewReprepareResponseHandler(ex));
                         return true;
                     });
                 return;
             }
-            Send(request, host, ReprepareResponseHandler);
+            Send(request, host, NewReprepareResponseHandler(ex));
         }
 
         /// <summary>
         /// Handles the response of a (re)prepare request and retries to execute on the same connection
         /// </summary>
-        private void ReprepareResponseHandler(IRequestError error, Response response, Host host)
+        private Action<IRequestError, Response, Host> NewReprepareResponseHandler(
+            PreparedQueryNotFoundException originalError)
         {
-            try
+            void ResponseHandler(IRequestError error, Response response, Host host)
```

Comment:
Does creating a new handler each time impact performance? I suspect that this will be optimized by the compiler, into some kind of closure.

Reply:
bah, nvm, this code is not within the hot path...